qml/Panel/Indicators/IndicatorsLight.qml

```diff
@@ -22,12 +23,126 @@ import Powerd 0.1
 import Lights 0.1
 import QMenuModel 0.1 as QMenuModel
 import Unity.Indicators 0.1 as Indicators
+import Wizard 0.1
 
 QtObject {
     id: root
 
     property color color: "darkgreen"
+    property int onMillisec: 1000
+    property int offMillisec: 3000
 
+    property double batteryLevel: 0
+    property string deviceState: ""
+    property bool hasMessages: false
+
+    property string batteryIconName: Status.batteryIcon
+    property string displayStatus: Powerd.status
+
+    onDisplayStatusChanged: {
+        updateLightState("onDisplayStatusChanged")
+    }
+
+    onBatteryIconNameChanged: {
+        updateLightState("onBatteryIconNameChanged")
+    }
+
+    function updateLightState(msg) {
+        console.log("updateLightState: " + msg + ", hasMessages: " + hasMessages + ", icon: " 
+            + batteryIconName + ", displayStatus: " + displayStatus + ", deviceState: " 
+            + deviceState + ", batteryLevel: " + batteryLevel)
+
+        // only show led when display is off
+        if(displayStatus == Powerd.On) {
+            Lights.state = Lights.Off
+            return
+        }
+
+        //
+        // priorities:
+        //   unread messsages (highest), full&charging, charging, low   
+        // 
+        // Icons: (see device.s from indicator-power)
+        //   %s-empty-symbolic             empty
+        //   %s-caution-charging-symbolic  charging  [ 0..10)
+        //   %s-low-charging-symbolic      charging  [10..30)
+        //   %s-good-charging-symbolic     charging  [30..60)
+        //   %s-full-charging-symbolic     charging  [60..100]
+        //   %s-full-charged-symbolic      fully charged
+        //   %s-low-symbolic               ?
+        //   %s-full-symbolic              ?
+        //
+        // device-state: (see system-settings\plugins\battery)
+        //   fully-charged
+        //   charging
+        //   discharging
+        //
+        // Indicators:
+        //   unread notifications : darkgreen pulsing (1000/3000)
+        //   charging             : white continuously
+        //   battery full         : green  continuously
+        //   battery low          : orangered pulsing (500/3000)
+        //
+        // Notes:
+        //   Icon name 'full-charged' comes late (45m after 100%)
+        //   so also check device-state and battery-level
+        //
+        //   Battery low warning dialog on screen shows up at 10%
+        //   but 'caution' icon at 9%.
+        // 
+
+        var lColor = ""
+        var lOnMS = -1
+        var lOffMS = -1
+
+        var isCharging = batteryIconName.indexOf("charging") >= 0
+        if(!isCharging)
```

Comment:
This is equivalent to

    var isCharging = (batteryIconName.indexOf("charging") >= 0) || (deviceState != "discharging)